route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -59,6 +59,12 @@ namespace route_following_plugin
             this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
         });
 
+        wml_->setMapCallback([this]() {
+            if (wm_->getRoute()) { // If this map update occured after a route was provided we need to regenerate maneuvers
+                this->latest_maneuver_plan_ = routeCb(wm_->getRoute()->shortestPath());
```

Comment:
Do we want a log statement here as well? To indicate what callback is causing the maneuver plan to be replanned

Reply:
Done